sdks/go/pkg/beam/core/runtime/exec/fn.go

```diff
@@ -224,15 +227,21 @@ func (n *invoker) Invoke(ctx context.Context, pn typex.PaneInfo, ws []typex.Wind
 }
 
 // ret1 handles processing of a single return value.
-// Errors or single values are the only options.
+// Errors, single values, or a ProcessContinuation are the only options.
 func (n *invoker) ret1(pn typex.PaneInfo, ws []typex.Window, ts typex.EventTime, r0 interface{}) (*FullValue, error) {
 	switch {
-	case n.outErrIdx >= 0:
+	case n.outErrIdx == 0:
 		if r0 != nil {
 			return nil, r0.(error)
 		}
 		return nil, nil
-	case n.outEtIdx >= 0:
+	case n.outPcIdx == 0:
+		if r0 == nil {
+			panic("invoker.ret1: cannot return a nil process continuation")
```

Comment:
It would be good to hydrate this panic (and our other panics) with information about which function is actually causing the problem

```suggestion
			panic(fmt.Sprintf("invoker.ret1: cannot return a nil process continuation from function %v", n.fn.Name()))
```